packages/mars/src/options/config.ts

```diff
@@ -30,28 +30,31 @@ export async function getConfig(options: Options): Promise<ExecuteOptions> {
     }
   }
 
-  const { wallet, networkName } = await getWallet(merged)
-  const gasPrice = merged.gasPrice ?? (await wallet.provider.getGasPrice())
+  const { signer, networkName } = await getSigner(merged)
+  const gasPrice = merged.gasPrice ?? (await signer.getGasPrice())
 
   return {
     gasPrice,
     noConfirm: !!merged.noConfirm,
-    wallet,
-    network: networkName,
+    signer,
+    networkName: networkName,
     dryRun: !!merged.dryRun,
     logFile: merged.logFile ?? '',
     deploymentsFile: merged.outputFile,
     verification,
   }
 }
 
-async function getWallet(options: Options) {
-  const { network, infuraApiKey, alchemyApiKey, dryRun, privateKey } = options
+async function getSigner(options: Options) {
+  const { network, infuraApiKey, alchemyApiKey, dryRun, fromAddress, privateKey } = options
   if (network === undefined) {
     throw new Error('No network specified. This should never happen.')
   }
   let rpcUrl
-  if (network.startsWith('http')) {
+  let provider
```

Comment:
Could you specify types for declared variables

Reply:
done